lib/cli/src/commands/compile.rs

```diff
@@ -83,6 +94,19 @@ impl Compile {
         println!("Target: {}", target.triple());
         let module = Module::from_file(&store, &self.path)?;
         let _ = module.serialize_to_file(&self.output)?;
+        #[cfg(feature = "object-file")]
+        if let Some(header_file_src) = module.artifact().create_header_file() {
+            // for C code
+            let mut header = std::fs::OpenOptions::new()
+                .create(true)
+                .truncate(true)
+                .write(true)
+                .open("test.h")?;
```

Comment:
I think it would be good to receive a filename from the CLI args rather than always using `test.h`